src/fields/fieldp256_scalar/ops.rs

```diff
@@ -0,0 +1,1782 @@
+  fiat_p256_scalar_addcarryx_u64(&mut x108, &mut x109, 0x0, x86, x100);
+  let mut x110: u64 = 0;
+  let mut x111: fiat_p256_scalar_u1 = 0;
+  fiat_p256_scalar_addcarryx_u64(&mut x110, &mut x111, x109, x88, x102);
+  let mut x112: u64 = 0;
+  let mut x113: fiat_p256_scalar_u1 = 0;
+  fiat_p256_scalar_addcarryx_u64(&mut x112, &mut x113, x111, x90, x104);
+  let mut x114: u64 = 0;
+  let mut x115: fiat_p256_scalar_u1 = 0;
+  fiat_p256_scalar_addcarryx_u64(&mut x114, &mut x115, x113, ((x91 as u64) + ((x85 as u64) + ((x77 as u64) + x65))), x106);
+  let x116: u64 = ((x115 as u64) + ((x107 as u64) + x95));
+  let mut x117: u64 = 0;
+  let mut x118: fiat_p256_scalar_u1 = 0;
+  fiat_p256_scalar_subborrowx_u64(&mut x117, &mut x118, 0x0, x110, 0xf3b9cac2fc632551);
+  let mut x119: u64 = 0;
+  let mut x120: fiat_p256_scalar_u1 = 0;
+  fiat_p256_scalar_subborrowx_u64(&mut x119, &mut x120, x118, x112, 0xbce6faada7179e84);
+  let mut x121: u64 = 0;
+  let mut x122: fiat_p256_scalar_u1 = 0;
+  fiat_p256_scalar_subborrowx_u64(&mut x121, &mut x122, x120, x114, 0xffffffffffffffff);
+  let mut x123: u64 = 0;
+  let mut x124: fiat_p256_scalar_u1 = 0;
+  fiat_p256_scalar_subborrowx_u64(&mut x123, &mut x124, x122, x116, 0xffffffff00000000);
+  let mut x125: u64 = 0;
+  let mut x126: fiat_p256_scalar_u1 = 0;
+  fiat_p256_scalar_subborrowx_u64(&mut x125, &mut x126, x124, (0x0 as u64), (0x0 as u64));
+  let mut x127: u64 = 0;
+  fiat_p256_scalar_cmovznz_u64(&mut x127, x126, x117, x110);
+  let mut x128: u64 = 0;
+  fiat_p256_scalar_cmovznz_u64(&mut x128, x126, x119, x112);
+  let mut x129: u64 = 0;
+  fiat_p256_scalar_cmovznz_u64(&mut x129, x126, x121, x114);
+  let mut x130: u64 = 0;
+  fiat_p256_scalar_cmovznz_u64(&mut x130, x126, x123, x116);
+  *IndexConst(&mut out1).index_mut(0) = x127;
+  *IndexConst(&mut out1).index_mut(1) = x128;
+  *IndexConst(&mut out1).index_mut(2) = x129;
+  *IndexConst(&mut out1).index_mut(3) = x130;
+}
+
+/// The function fiat_p256_scalar_to_bytes serializes a field element NOT in the Montgomery domain to bytes in little-endian order.
+///
+/// Preconditions:
+///   0 ≤ eval arg1 < m
+/// Postconditions:
+///   out1 = map (λ x, ⌊((eval arg1 mod m) mod 2^(8 * (x + 1))) / 2^(8 * x)⌋) [0..31]
+///
+/// Input Bounds:
+///   arg1: [[0x0 ~> 0xffffffffffffffff], [0x0 ~> 0xffffffffffffffff], [0x0 ~> 0xffffffffffffffff], [0x0 ~> 0xffffffffffffffff]]
+/// Output Bounds:
+///   out1: [[0x0 ~> 0xff], [0x0 ~> 0xff], [0x0 ~> 0xff], [0x0 ~> 0xff], [0x0 ~> 0xff], [0x0 ~> 0xff], [0x0 ~> 0xff], [0x0 ~> 0xff], [0x0 ~> 0xff], [0x0 ~> 0xff], [0x0 ~> 0xff], [0x0 ~> 0xff], [0x0 ~> 0xff], [0x0 ~> 0xff], [0x0 ~> 0xff], [0x0 ~> 0xff], [0x0 ~> 0xff], [0x0 ~> 0xff], [0x0 ~> 0xff], [0x0 ~> 0xff], [0x0 ~> 0xff], [0x0 ~> 0xff], [0x0 ~> 0xff], [0x0 ~> 0xff], [0x0 ~> 0xff], [0x0 ~> 0xff], [0x0 ~> 0xff], [0x0 ~> 0xff], [0x0 ~> 0xff], [0x0 ~> 0xff], [0x0 ~> 0xff], [0x0 ~> 0xff]]
+#[inline]
+pub const fn fiat_p256_scalar_to_bytes(mut out1: &mut [u8; 32], arg1: &[u64; 4]) {
```

Comment:
Should we add an `impl Codec` for this? Because `FieldP256` has `impl Codec` and `CodecFieldElement`.